cfme/tests/cli/test_appliance_update.py

```diff
@@ -33,20 +33,20 @@ def pytest_generate_tests(metafunc):
 
     split_ver = str(version).split(".")
     try:
-        minor_build = split_ver[2]
-        assert int(minor_build) != 0
-    except IndexError:
-        logger.exception('Caught IndexError generating for test_appliance_update, skipping')
-    except AssertionError:
-        logger.debug('Caught AssertionError: No previous z-stream version to update from')
+        minor_build = int(split_ver[2])
+    except (IndexError, ValueError) as e:
+        logger.exception("Couldn't parse version: %s, skipping", e)
         versions.append(pytest.param("bad:{!r}".format(version), marks=pytest.mark.uncollect(
-            'Could not parse minor_build version from: {}'.format(version)
+            reason='Could not parse minor_build version from: {}'.format(version)
         )))
     else:
-        for i in range(int(minor_build) - 1, -1, -1):
-            # removing older 5.9 builds due to sprout limitation.
-            if version < '5.10' and i > 1:
-                versions.append("{}.{}.{}".format(split_ver[0], split_ver[1], i))
+        if minor_build < 1:
+            logger.debug('No previous z-stream version to update from.')
+            versions.append(pytest.param("bad:{!r}".format(version), marks=pytest.mark.uncollect(
+                reason='No previous z-stream version to update from: {}'.format(version)
+            )))
+        else:
+            versions.append("{}.{}.{}".format(split_ver[0], split_ver[1], minor_build - 1))
     metafunc.parametrize('old_version', versions, indirect=True)
```

Comment:
It would be better if we format with **key-value** pair.
example:
 ```versions.append("{key1}.{key2}.{key3}".format(key1=split_ver[0], key2=split_ver[1], key3=minor_build - 1)```

Reply:
I am not so sure about that. How do you think I should name the version parts?

Reply:
I don't think the example you provided is better than what is currently there. I think it is just a matter of personal taste.